src/Maestro/Microsoft.DotNet.Maestro.Tasks/src/PushMetadataToBuildAssetRegistry.cs

```diff
@@ -410,15 +410,20 @@ private BuildData MergeBuildManifests(List<BuildData> buildsMetadata)
                     mergedBuild.Commit != build.Commit ||
                     mergedBuild.AzureDevOpsRepository != build.AzureDevOpsRepository)
                 {
-                    throw new Exception("Can't merge if one or more manifests have different branch, build number, commit or repository values.");
+                    throw new Exception("Can't merge if one or more manifests have different branch, build number, commit, or repository values.");
                 }
 
                 mergedBuild.Assets = mergedBuild.Assets.AddRange(build.Assets);
             }
 
-            // Remove duplicated assets based on the top level properties of the asset.
-            // The AssetLocations property isn't set at this point yet.
-            mergedBuild.Assets = mergedBuild.Assets.Distinct(new AssetDataComparer()).ToImmutableList();
+            // Error out for any duplicated assets based on the top level properties of the asset.
+            var distinctAssets = mergedBuild.Assets.Distinct(new AssetDataComparer()).ToImmutableList();
+            if (distinctAssets.Count < mergedBuild.Assets.Count)
```

Comment:
Any chance we can inform which ones are duplicated to save on investigation time?

Reply:
oh sure that shouldn't be hard

Reply:
NIT: I think the duplication will most likely occur because different build manifests have same asset listed. So it would be helpful to say which manifest(s) contains the asset. I don't think that's a requirement for the PR, though.